xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -524,6 +525,7 @@ void CSelectionStreams::Update(const std::shared_ptr<CDVDInputStream>& input,
         CDemuxStreamVideo* vstream = static_cast<CDemuxStreamVideo*>(stream);
         s.width = vstream->iWidth;
         s.height = vstream->iHeight;
+        s.aspect_ratio = vstream->fAspect;
```

Comment:
I also once created a PR with the exact same fix but it was never merged because of comments by @FernetMenta . Please see https://github.com/xbmc/xbmc/pull/16239

Reply:
It already should be by the if statement in `CApplication.cpp`.  We're only updating stuff using vp that doesn't already have any details and even then, only if it's from DVD or Bluray. Everything else is updated by the background loader isn't it ? I've tested this by removing everything from the streamdetails table in the db then playing something not dvd related that the background loader hasn't got to yet and the details were not updated from vp.

Reply:
Ok, then I guess we should be ok.